src/transformers/models/plbart/modeling_plbart.py

```diff
@@ -0,0 +1,1734 @@
+        if isinstance(module, (PLBartDecoder, PLBartEncoder)):
+            module.gradient_checkpointing = value
+
+    @property
+    def dummy_inputs(self):
+        pad_token = self.config.pad_token_id
+        input_ids = torch.tensor([[0, 6, 10, 4, 2], [0, 8, 12, 2, pad_token]], device=self.device)
+        dummy_inputs = {
+            "attention_mask": input_ids.ne(pad_token),
+            "input_ids": input_ids,
+        }
+        return dummy_inputs
+
+
+PLBART_START_DOCSTRING = r"""
+    This model inherits from [`PreTrainedModel`]. Check the superclass documentation for the generic methods the
+    library implements for all its model (such as downloading or saving, resizing the input embeddings, pruning heads
+    etc.)
+
+    This model is also a PyTorch [torch.nn.Module](https://pytorch.org/docs/stable/nn.html#torch.nn.Module) subclass.
+    Use it as a regular PyTorch Module and refer to the PyTorch documentation for all matter related to general usage
+    and behavior.
+
+    Parameters:
+        config ([`PLBartConfig`]):
+            Model configuration class with all the parameters of the model. Initializing with a config file does not
+            load the weights associated with the model, only the configuration. Check out the
+            [`~PreTrainedModel.from_pretrained`] method to load the model weights.
+"""
+
+PLBART_GENERATION_EXAMPLE = r"""
+    Token in-filling example::
+
+        >>> from transformers import PLBartTokenizer, PLBartForConditionalGeneration, PLBartConfig
+
+        >>> model = PLBartForConditionalGeneration.from_pretrained('uclanlp/plbart-base') >>> tokenizer =
+        PLBartTokenizer.from_pretrained('uclanlp/plbart-base', src_lang='java', tgt_lang='java')
+
+        >>> METHOD_TO_FILL = "public static main (String args[0]) { data=Date(); System.out. String.format("Current
+        Date : % tc", ));}" >>> inputs = tokenizer([METHOD_TO_FILL], max_length=1024, return_tensors='pt')
+
+        >>> # Generate Filled Code >>> generated_ids = model.generate(inputs['input_ids'], num_beams=4, max_length=5,
+        early_stopping=True) >>> print([tokenizer.decode(g, skip_special_tokens=True,
+        clean_up_tokenization_spaces=False) for g in generated_ids])
+
+    Mask-filling example::
+
+        >>> from transformers import PLBartTokenizer, PLBartForConditionalGeneration >>> tokenizer =
+        PLBartTokenizer.from_pretrained('uclanlp/plbart-base') >>> # en_XX is the language symbol id <LID> for English
+        >>> TXT = "</s> Is 0 the <mask> Fibonacci <mask> ? </s> en_XX"
+
+        >>> model = PLBartForConditionalGeneration.from_pretrained('uclanlp/plbart-base') >>> input_ids =
+        tokenizer([TXT], add_special_tokens=False, return_tensors='pt')['input_ids'] >>> logits =
+        model(input_ids).logits
+
+        >>> masked_index = (input_ids[0] == tokenizer.mask_token_id).nonzero().item() >>> probs = logits[0,
+        masked_index].softmax(dim=0) >>> values, predictions = probs.topk(5)
```

Comment:
Those examples need fixing, and tehre should only be one colon after example. We don't use rst anymore.